scripts/proto-gen.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+#
+# Update the generated code for protocol buffers in the Tendermint repository.
+# This must be run from inside a Tendermint working directory.
+#
+set -euo pipefail
+
+# Work from the root of the repository.
+cd "$(git rev-parse --show-toplevel)"
+
+# Run inside Docker to install the correct versions of the required tools
+# without polluting the local system.
+docker run --rm -i -v "$PWD":/w --workdir=/w golang:1.18-alpine sh <<"EOF"
+apk add curl git make
+
+readonly buf_release='https://github.com/bufbuild/buf/releases/latest/download'
+readonly OS="$(uname -s)" ARCH="$(uname -m)"
+curl -sSL "${buf_release}/buf-${OS}-${ARCH}.tar.gz" \
+        | tar -xzf -  -C /usr/local --strip-components=1
```

Comment:
I feel mixed about leaking binaries into /usr/local/ for potentially chaning tarballs/tarbombs. is it viable to just put it in `<reop>/build`?

Reply:
I completely agree. But here we're in a throwaway container, so it doesn't matter.